lib/internal/modules/esm/resolve.js

```diff
@@ -907,12 +936,15 @@ function moduleResolve(specifier, base, conditions, preserveSymlinks) {
     try {
       resolved = new URL(specifier);
     } catch {
-      resolved = packageResolve(specifier, base, conditions);
+      ({ resolved, format } = packageResolve(specifier, base, conditions));
     }
   }
   if (resolved.protocol !== 'file:')
-    return resolved;
-  return finalizeResolution(resolved, base, preserveSymlinks);
+    return { url: resolved };
+  return {
+    url: finalizeResolution(resolved, base, preserveSymlinks),
+    ...(format !== undefined && format !== null) && { format: format }
+  };
```

Comment:
I think we should add brackets, I'm not sure the suggestion above really improves readability, having brackets should remove any ambiguity.

Reply:
Sure, or that. Just the indented version with nothing was my concern 🙂

Reply:
I agree with the readability argument and also it is IMO good practice to have curly braces also on one-statement if blocks. This was like this in existing code and there is a number of places where it is following the same pattern. Did not want to start reformatting too much here. But since we are at this point in code I think it is good to fix it here.